app.js

```diff
@@ -1,16 +1,61 @@
 "use strict";
 
-var express = require('express');
-var mongoose = require('mongoose');
+var http = require('http'),
+    domain = require('domain'),
+    express = require('express'),
+    mongoose = require('mongoose'),
+    _ = require('underscore'),
+    pg = require('pg'),
+    bodyParser = require('body-parser'),
+    prettify = require('./prettify.js')(),
+    db = require('./db.js')(pg, prettify),
+    Session = require('./entities/Session.js')(_, pg, db),
+    User = require('./entities/User.js')(Session, pg, db),
+    Event = require('./entities/Event.js')(pg, db),
+    Project = require('./entities/Project.js')(pg, db),
+    Donate = require('./entities/Donate.js')(pg, db),
+    methods = require('./methods.js')(_, pg, User, Session, Event, Project, Donate),
+    app = express();
 
-var app = express();
-
-mongoose.connect(process.env.MONGODB_URI);
+mongoose.connect(process.env.MONGODB_URI || "mongo://uvs@localhost");
 
 app.set('port', (process.env.PORT || 3000));
 
 require('./routes.js')(app);
 
-app.listen(app.get('port'), function () {
-    console.log('Example app listening on port', app.get('port'));
-});
\ No newline at end of file
+// app.use(cookieParser());
+app.use(bodyParser.json({
+    limit: '150mb'
+})); // for parsing application/json
+// app.use(bodyParser.urlencoded({ extended: true }));
+
+app.use(express.static('public'));
+
+app.post('/*', function (req, res) {
+    var methodName = req.url.split('/')[1]
+    console.log(methodName, req.body);
+    Session.restoreSession(req.body.sessionId, function(err, session) {
+        if(err){
+            res.status(400).json({err})
+        } else {
+            if(!_.isObject(methods[session.userPriviledge]) || !_.isFunction(methods[session.userPriviledge][methodName])){
+                res.status(400).json({err: {
+                    code: 2,
+                    message: 'Access denied'
```

Comment:
Large vendors had a lot of legacy, inner protocol restrictions, and resources to manage, maintain and support their own standards. If we can fit with public standards like [HTTP 1.1 RFC7231](https://tools.ietf.org/html/rfc7231) we must.